iothub/service/src/Common/Data/SharedAccessSignatureAuthorizationRule.cs

```diff
@@ -1,18 +1,25 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
 using System;
 using Newtonsoft.Json;
 
 namespace Microsoft.Azure.Devices.Common.Data
 {
-    public sealed class SharedAccessSignatureAuthorizationRule : IEquatable<SharedAccessSignatureAuthorizationRule>
+    /// <summary>
+    /// A shared access signature based authorization rule for authenticating requests against an IoT Hub.
+    /// </summary>
+    internal sealed class SharedAccessSignatureAuthorizationRule : IEquatable<SharedAccessSignatureAuthorizationRule>
```

Comment:
I made this class internal since it was not being referenced anywhere in our public APIs -> users could have initialized this class, but there was never any way that they could pass this to our client.

Reply:
I see it being used here:
https://github.com/Azure/iotedge/blob/2d26602a08c787551ce9f6dc977bab3169048a7d/edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/authenticators/DeviceScopeTokenAuthenticator.cs

Reply:
actually, this might be a different SharedAccessSignatureAuthorizationRule...

Reply:
that is a good find!
This being a service client class, I'm surprised that the edge team is taking a dependency on this, but I'll definitely verify it with them before merging the PR in.

Reply:
This is sooo weird. How can we move these to the Edge team :D We should not have helpers for their code. I saw a similar thing on my PR.

Reply:
So, they fork our repo and add some stuff on top of it, which they then version it as the "Nested-Edge-xxx" in their code (this is not publicly available). I am planning to send across an API surface diff to Varun to let him know of our intended changes. If you'd like I can generate the diff for your PR files as well @vinagesh .